### PR TITLE
Feature/pisa25 376/FLA mediaInteraction authoring properties

### PR DESCRIPTION
**Pull request by oatymart**
https://oat-sa.atlassian.net/browse/PISA25-376
https://oat-sa.atlassian.net/browse/PISA25-377

## Changes to MediaInteraction
- when authored with audio type, and `autostart` is enabled, these subproperties appear:
  - `attr: data-autostart-delay-ms` (shown to user in seconds) - disabled by default. 0-600 range.
  - `class: hide-player` (shown to user as "Display media player") - on by default, off enables delay and dims player widget and disables pause
  - `class: sequential` - off by default; on disables loop and limits maxPlays to 0 or 1

Screenshot 2023-06-29 at 11 20 49

_Changes were done by re-rendering the handlerbars template every time, which is not usually done and can cause some flickering. I think it's acceptable though._

## Not supported
- Most of the new properties won't work in Terre item preview (since only Solar delivery engine will have support). This will just have to be explained to the customer, in case they are still using TAO with Terre previews.

## To do
- Feature flag to be added, so these options are invisible to most customers
- Unit test? But I did not find any for any other interaction in qtiCreator, so it seems unlikely...

## To test
XML debugger TAO extension is very useful for checking interaction XML.
Deployed version: https://delivery.test-pisa378.playground.kitchen.it.taocloud.org/

### CODE REVIEW

**Review by wazelin — CHANGES_REQUESTED** [review]

File: views/js/qtiCreator/tpl/forms/interactions/media.tpl [hunk 1]
```diff
@@ -8,11 +7,12 @@
         </label>
     </div>
 
-    <h3 class="media-sizer-panel-label">{{__ 'Size and position'}}</h3>
-    <div class="panel media-sizer-panel">
-        <!-- mediaEditorComponent goes here -->
-    </div>
-
+    {{#unless isAudio}}
```
[the remark this answers is not in the document]
Comment (wazelin) [reply views/js/qtiCreator/tpl/forms/interactions/media.tpl:10]: Oh. OK. The reason there is that the widget gets re-initialized on the first data change, but the callbacks are not getting applied to the newly initialized widget.

I've addressed this [in my PR here](https://github.com/oat-sa/extension-tao-itemqti/pull/2382/files/02b2910b3f7938260f14409dfc467b4aef19dafa..64202cf600402fe5a1965958682558650d217380).

File: views/js/qtiCreator/tpl/forms/interactions/media.tpl [hunk 2]
```diff
@@ -8,11 +7,12 @@
...
+        <h3 class="media-sizer-panel-label">{{__ 'Size and position'}}</h3>
+        <div class="panel media-sizer-panel">
+            <!-- mediaEditorComponent goes here -->
```
Comment (olga-kulish) [inline views/js/qtiCreator/tpl/forms/interactions/media.tpl:13]: 1. Create new interaction
2. Upload _video_
3. Check 'Autostart' ---> Size slider dissapears and doesn't reappear when you uncheck it

(I'm testing this branch: https://github.com/oat-sa/extension-tao-itemqti/pull/2382)

Comment (oatymart) [reply views/js/qtiCreator/tpl/forms/interactions/media.tpl:13]: Fix is in progress...

File: model/FeatureFlag/FeatureFlagFlaHandler.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+namespace oat\taoQtiItem\model\FeatureFlag;
+
+use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
+use oat\tao\model\featureFlag\FeatureFlagConfigHandlerInterface;
+
+class FeatureFlagFlaHandler implements FeatureFlagConfigHandlerInterface
+{
+    private FeatureFlagCheckerInterface $featureFlagChecker;
+
+    public function __construct(FeatureFlagCheckerInterface $featureFlagChecker) {
```
Comment (olga-kulish) [inline model/FeatureFlag/FeatureFlagFlaHandler.php:34]: CI complains https://github.com/oat-sa/extension-tao-itemqti/actions/runs/5443164884/jobs/9899343282?pr=2381
Screenshot_13

**Review by olga-kulish — COMMENTED** [review]
All good for me. I'll approve if no other changes are expected.

File: model/FeatureFlag/FeatureFlagFlaHandler.php (hunk 3 above)
Comment (wazelin) [reply model/FeatureFlag/FeatureFlagFlaHandler.php:34]: I'm on it.

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by wazelin — APPROVED** [review]

**Review by wazelin — APPROVED** [review]

**Pull request merged**